File: phase-1-arrays/LC424/code.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int l = 0 , maxCount = 0 , result = 0 ; 
        vector<int> count(26 , 0); 
        for(int r=  0 ; r < s.size() ; r++)
        {
            count[s[r] - 'A']++; 
            maxCount = max(maxCount ,  count[s[r] - 'A']); 
            while((r - l + 1) - maxCount  > k)
            {
                count[s[l] - 'A']--; 
                l++; 
            } 
            result = max(result , r - l + 1); 
        }
        return result ; 
    }
}; 

File: phase-1-arrays/LC424/code.cs
public class Solution
{
    public int CharacterReplacement(string s, int k)
    {
        int[] count = new int[26];

        int left = 0;
        int maxCount = 0;
        int result = 0;

        for (int right = 0; right < s.Length; right++)
        {
            count[s[right] - 'A']++;

            maxCount = Math.Max(
                maxCount,
                count[s[right] - 'A']
            );

            while ((right - left + 1) - maxCount > k)
            {
                count[s[left] - 'A']--;
                left++;
            }

            result = Math.Max(
                result,
                right - left + 1
            );
        }

        return result;
    }
}